methods/eps_fhir/api_methods.py

```diff
@@ -0,0 +1,125 @@
+import json
+import uuid
+
+from features.environment import CIS2_USERS
+from methods.eps_fhir.api_request_body_generators import (
+    create_fhir_bundle,
+    create_fhir_parameter,
+    generate_message_header,
+    generate_medication_request,
+    generate_patient,
+    generate_organization,
+    generate_practitioner_role,
+    generate_practitioner,
+    generate_provenance,
+    generate_agent,
+    generate_owner,
+    generate_group_identifier,
+)
+from methods.shared.common import the_expected_response_code_is_returned
+from methods.shared.api import post, get_default_headers
+from utils.prescription_id_generator import generate_short_form_id
+from utils.signing import get_signature
+
+
+def create_new_prepare_body(context):
+    context.sender_ods_code = "A83008"
```

Comment:
Not all of these need adding to context.

Reply:
Which one we should avoid adding to the context?
I believe they are all used in the various stages of the prescription and for both nominated and non-nominated.

Reply:
Looks like sender_ods_code, prescription_item_id, long_prescription_id and secondary_care_type are only used from this function. Only receiver_ods_code and prescription_id are retrieved from the context in others.

Reply:
I believe I see what you mean. So, the generated body from the prepare step is carried all the way to the singing step.
if you notice we use `context.prepare_body = _create_new_prepare_body(context)` in prepare_prescription and then `body = json.loads(context.prepare_body)` in the `_create_signed_body` because they are identical, only with the addition of provenance. So by passing everything onto the context at the first step we are able to carry them away. 

We were trying to generate/hardcode them prior which was the reason everything was breaking. 

Does that make more sense?

Reply:
These will need adding to context at this point when we come to doing data assertion in later steps. You're right in the fact they're not used right now as we are only "indicating success" but were left in for this reason.

Reply:
This has been changed as suggested, context items may return in the future when we need them for later assertions